### PR TITLE
ci: Create buildspec.yml for CodeBuild deployments

### PR DESCRIPTION
**Pull request by luislhl**
### Acceptance Criteria
- CodeBuild should be capable of knowing which branch/tag it should deploy
- It should run migrations and deploy the Lambdas to the right environment, according to the branch/tag
- The docs should be updated explaining how to run deploys and how they work

### CODE REVIEW

File: docs/2021-07-29-infrastructure-design.md [hunk 1]
```diff
@@ -62,13 +62,22 @@ Daemon's deploy will consist of:
...
+### How the process works
+We use CodeBuild and CodePipeline in conjunction, in a series of steps.
+
+Those are the steps that happen when a commit or tag is created in Github:
+
+1. A Github Webhook in the repo sends a request to one of our CodeBuild projects, that we will call [github-trigger](https://eu-central-1.console.aws.amazon.com/codesuite/codebuild/769498303037/projects/hathor-wallet-service-github-trigger/history).
+2. This project checks the branch and the author of the commit, to decide if it should run the build. If so, it runs some steps that are defined inline in the project, to write to a file which branch is being deployed, and uploads the project's code and this file to an S3 bucket: https://eu-central-1.console.aws.amazon.com/codesuite/codebuild/projects/hathor-wallet-service-github-trigger/edit/buildspec?region=eu-central-1
+3. A CodePipeline project is listening to changes in the bucket, and triggers when it detects them.
+4. CodePipeline calls its next step, which is asking for Manual Approval. A message is sent to Slack channel `#deploys`.
```
Comment (andreabadesso) [inline docs/2021-07-29-infrastructure-design.md:77]: That is cool. Will there also be a button for approval on Slack?

If not, where do I approve the deploy?

File: .codebuild/buildspec.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        if expr "${GIT_REF_TO_DEPLOY}" : "v.*" >/dev/null; then
+          # Gets all env vars with `mainnet_` prefix and re-exports them without the prefix
+          for var in "${!mainnet_@}"; do
+            export ${var#mainet_}=${!var}
+          done
+
+          make migrate;
+          make deploy-lambdas-mainnet;
+        fi
+  #post_build:
```
Comment (andreabadesso) [inline .codebuild/buildspec.yml]: Remove?

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: docs/2021-07-29-infrastructure-design.md (hunk 1 above)
Comment (luislhl) [reply docs/2021-07-29-infrastructure-design.md:77]: You click the link on Slack that will lead you to a CodePipeline page with the button.

The `#deploys` channel already exists, you can check the examples there.

File: .codebuild/buildspec.yml (hunk 2 above)
Comment (luislhl) [reply .codebuild/buildspec.yml]: Done

File: README.md [hunk 3]
```diff
@@ -200,3 +200,5 @@ Sometimes, jest will use old cached js files, even after you modified the typesc
 ```
 ./node_modules/.bin/jest --clearCache
 ```
+
+## Standard Operating Procedures
```
Comment (pedroferreira1) [inline README.md]: Should this be removed? Did you start here and moved to the other file?

File: docs/2021-07-29-infrastructure-design.md [hunk 4]
```diff
@@ -62,13 +62,22 @@ Daemon's deploy will consist of:
...
 
-The way it works is similar to Github Actions. We create a spec file declaring the steps that we want it to run, and configuring which branch we want to trigger the build. It seems to be possible to trigger it on GitHub releases too.
+### How the process works
+We use CodeBuild and CodePipeline in conjunction, in a series of steps.
+
+Those are the steps that happen when a commit or tag is created in Github:
+
+1. A Github Webhook in the repo sends a request to one of our CodeBuild projects, that we will call [github-trigger](https://eu-central-1.console.aws.amazon.com/codesuite/codebuild/769498303037/projects/hathor-wallet-service-github-trigger/history).
+2. This project checks the branch and the author of the commit, to decide if it should run the build. If so, it runs some steps that are defined inline in the project, to write to a file which branch is being deployed, and uploads the project's code and this file to an S3 bucket: https://eu-central-1.console.aws.amazon.com/codesuite/codebuild/projects/hathor-wallet-service-github-trigger/edit/buildspec?region=eu-central-1
```
Comment (pedroferreira1) [inline docs/2021-07-29-infrastructure-design.md:75]: Why do you check the author of the commit? Depending on the author you do different things?

**Review by pedroferreira1 — APPROVED** [review]

File: docs/2021-07-29-infrastructure-design.md (hunk 4 above)
Comment (luislhl) [reply docs/2021-07-29-infrastructure-design.md:75]: Only specific authors will trigger a deploy. A commit from someone that is not in the list will not trigger a deploy.

This is a built-in feature of CodeBuild and is explained better in the first best practice described in https://docs.aws.amazon.com/codebuild/latest/userguide/webhooks.html?icmpid=docs_acb_console#webhook-best-practices

File: README.md (hunk 3 above)
Comment (luislhl) [reply README.md]: There should be a link here to the other file, I'll fix this.

Comment (luislhl) [reply README.md]: Done

**Review by andreabadesso — APPROVED** [review]

**Pull request merged**